source/UberCore/Authentication/OAuthEndpoint.swift

```diff
@@ -76,27 +87,62 @@ public enum OAuth: APIEndpoint {
             return "/oauth/v2/authorize"
         case .refresh:
             return "/oauth/v2/mobile/token"
+        case .par:
+            return "/oauth/v2/par"
         }
     }
 
     public var query: [URLQueryItem] {
         switch self {
-        case .implicitLogin(let clientID, let scopes, let redirect):
+        case .implicitLogin(let clientID, let scopes, let redirect, let requestUri):
             var loginQuery = baseLoginQuery(clientID, redirect: redirect, scopes: scopes)
-            let additionalQueryItems = queryBuilder(("response_type", "token"))
+            let additionalQueryItems: [URLQueryItem] = [
+                ("response_type", ResponseType.token.rawValue),
+                ("request_uri", requestUri)
+            ]
+            .compactMap { pair -> [URLQueryItem]? in
+                guard let value = pair.1 else {
+                    return nil
+                }
+                return self.queryBuilder((pair.0, value))
+            }
+            .flatMap { $0 }
 
             loginQuery.append(contentsOf: additionalQueryItems)
             return loginQuery
-        case .authorizationCodeLogin(let clientID, let redirect, let scopes, let state):
+        case .authorizationCodeLogin(let clientID, let redirect, let scopes, let state, let requestUri):
             var loginQuery = baseLoginQuery(clientID, redirect: redirect, scopes: scopes)
-            let additionalQueryItems = queryBuilder(("response_type", "code"),
-                                                    ("state", state ?? ""))
+            let additionalQueryItems: [URLQueryItem] = [
+                ("response_type", ResponseType.code.rawValue),
+                ("state", state ?? ""),
```

Comment:
This logic looks very similar to line 99->109. Maybe it can be streamlined a bit.